cloud/scope/powervs_cluster_test.go

```diff
@@ -3830,3 +3832,799 @@ func TestCreateVPCSubnet(t *testing.T) {
+	})
+	t.Run("When GetDHCPServer returns error", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		clusterScope := PowerVSClusterScope{
+			IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+				Status: infrav1beta2.IBMPowerVSClusterStatus{
+					DHCPServer: &infrav1beta2.ResourceReference{
+						ID:                ptr.To("dhcpServerID"),
+						ControllerCreated: ptr.To(true),
+					},
+					ServiceInstance: &infrav1beta2.ResourceReference{},
+				},
+			},
+			IBMPowerVSClient: mockPowerVS,
+		}
+		mockPowerVS.EXPECT().GetDHCPServer(gomock.Any()).Return(nil, fmt.Errorf("error getting dhcp server"))
+		err := clusterScope.DeleteDHCPServer()
+		g.Expect(err.Error()).To(Equal("failed to fetch DHCP server: error getting dhcp server"))
+	})
+	t.Run("When DeleteDHCPServer returns error", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		clusterScope := PowerVSClusterScope{
+			IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+				Status: infrav1beta2.IBMPowerVSClusterStatus{
+					DHCPServer: &infrav1beta2.ResourceReference{
+						ID:                ptr.To("dhcpServerID"),
+						ControllerCreated: ptr.To(true),
+					},
+					ServiceInstance: &infrav1beta2.ResourceReference{},
+				},
+			},
+			IBMPowerVSClient: mockPowerVS,
+		}
+		dhcpServer := &models.DHCPServerDetail{ID: ptr.To("dhcpServerID")}
+		mockPowerVS.EXPECT().GetDHCPServer(gomock.Any()).Return(dhcpServer, nil)
+		mockPowerVS.EXPECT().DeleteDHCPServer(gomock.Any()).Return(fmt.Errorf("error deleting dhcp server"))
+		err := clusterScope.DeleteDHCPServer()
+		g.Expect(err.Error()).To(Equal("failed to delete DHCP server: error deleting dhcp server"))
+	})
+	t.Run("When DHCP server deletion is successful", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		clusterScope := PowerVSClusterScope{
+			IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+				Status: infrav1beta2.IBMPowerVSClusterStatus{
+					DHCPServer: &infrav1beta2.ResourceReference{
+						ID:                ptr.To("dhcpServerID"),
+						ControllerCreated: ptr.To(true),
+					},
+					ServiceInstance: &infrav1beta2.ResourceReference{},
+				},
+			},
+			IBMPowerVSClient: mockPowerVS,
+		}
+		dhcpServer := &models.DHCPServerDetail{ID: ptr.To("dhcpServerID")}
+		mockPowerVS.EXPECT().GetDHCPServer(gomock.Any()).Return(dhcpServer, nil)
+		mockPowerVS.EXPECT().DeleteDHCPServer(gomock.Any()).Return(nil)
+		err := clusterScope.DeleteDHCPServer()
+		g.Expect(err).To(BeNil())
+	})
+}
+
+func TestDeleteTransitGatewayConnections(t *testing.T) {
+	var (
+		mockTransitGateway *tgmock.MockTransitGateway
+		mockCtrl           *gomock.Controller
+	)
+	setup := func(t *testing.T) {
+		t.Helper()
+		mockCtrl = gomock.NewController(t)
+		mockTransitGateway = tgmock.NewMockTransitGateway(mockCtrl)
+	}
+	teardown := func() {
+		mockCtrl.Finish()
+	}
+	t.Run("When transit gateway in PowerVS is in deleting state", func(t *testing.T) {
```

Comment:
```suggestion
	t.Run("When PowerVS connection of transit gateway is in deleting state", func(t *testing.T) {
```
The test case name might cause confusion, please change it wherever applicable for this function to specify as `PVS connection or VPC connection of a TG`